lib/system/memalloc.nim

```diff
@@ -33,6 +33,19 @@ when notJSnotNims:
     ## otherwise. Like any procedure dealing with raw memory this is
     ## **unsafe**.
 
+  proc cmpMem*(a, b: pointer, size: Natural): int {.inline, noSideEffect,
```

Comment:
```suggestion
  func cmpMem*(a, b: pointer, size: Natural): int {.inline,
```

Reply:
Because only two functions with `proc + noSideEffect`, I recommend to change them in this PR.

https://github.com/nim-lang/Nim/blob/a6c80b3920582b3b62ac821436d739c03352e210/lib/system/memalloc.nim#L27

Reply:
I'm not sure if this is correct

Reply:
see https://github.com/nim-lang/Nim/pull/16281

Reply:
Yeah, because it has function declaration, maybe don't change them.